unity/Holo/Assets/Scripts/STLConversion/STLSeriesImporter.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+//A class for importing a STL series from a dir
+public class STLSeriesImporter
+{
+    GameObject importedSTLSeries = new GameObject();
+    Mesh mesh = new Mesh();
+    private string[] filePaths;
+    public string FileName { get; private set;}
+    public GameObject GetGameObject()
+    {
+        return importedSTLSeries;
+    }
+
+    public Mesh GetMesh()
+    {
+        return mesh;
+    }
+    // Object constructor, initiates STL series import.
+    public STLSeriesImporter()
+    {
+        GetFilePaths();
+        GetFilename();
+        LoadFiles();
+    }
+
+
+    // gets path to the subsequent STL meshes stored in a root folder.
+    private void GetFilePaths()
+    {
+        string rootFolder = EditorUtility.OpenFolderPanel("Select STL series root folder", Application.dataPath, "");
+        filePaths = Directory.GetFiles(rootFolder + @"\");
+    }
+
+    private void GetFilename()
+    {
+        string fileName = Path.GetFileNameWithoutExtension(filePaths[0]);
+
+        while (true)
+        {
+            char lastChar = fileName[fileName.Length - 1];
+            char endChar = '-';
+            fileName = fileName.Remove(fileName.Length - 1);
+            if (lastChar == endChar)
+            {
+                break;
+            }
+        }
```

Comment:
This is not secured in case String doesn't contain "-" anywhere.

I would instead use one call to `String.LastIndexOf` to find the last "-" ( https://docs.microsoft.com/en-us/dotnet/api/system.string.lastindexof?view=netframework-4.7.2 ), and then one call to `String.Substring` to get the initial part of the string.

Reply:
Thanks and done! :)